### DISCUSSION TITLE
Java client help please

### DISCUSSION THREAD

**Discussion by Pseudow** [seaweedfs/seaweedfs#4900] [Q&A]
Hi I am kinda new to SeaweedFS and I wondered how can I using the Java client with SeaweedOutputStream how can I specify metadata and should I use a database for the metadata store with the fid?

**Comment by chrislusf** [comment]
what kind of metadata?

**Comment by Pseudow** [comment]
I would like to store additional data such as The author and some of its information but also some others like labels or description or a specific id. I was planning on storing it on my database but I can't find any fid to refer to the file, should I use its path?

**Reply by chrislusf** [reply]
For this purpose, the SeaweedOutputStream class would need an additional API to set the additional attributes in the `extended` attribute map, which is currently not exposed.

**Reply by Pseudow** [reply]
Okay thank you, do you have any idea on when this feature would be implemented? I think I am just going to use the filer's path to store a certain kind of unique id to this file.

**Comment by Barfoolozie** [comment]
SeaweedFS is like a big digital storage place for computer files.

1: First, you need to get a special tool (like a helper) to put your files into SeaweedFS. This tool works with Java, a computer language.

2: You tell this tool to add some extra notes about your file before putting it into SeaweedFS.

3: You can also use a different tool (like a notebook) to write more details about your file, like what it's about or who made it.

4: Some people use a "digital notebook" called a database to keep these extra details safe.

5: When you want to find your files and their notes later, the database helps you do that easily.

It's like putting your toys in a box, writing what each toy is on a sticky note, and using a special notebook to remember more about your toys.

If you need help with any of the 5 points I talked about above just send me a message or reply here id be more than happy to help you out.
